packages/libarchive.cmake

```diff
@@ -6,24 +6,39 @@ ExternalProject_Add(libarchive
         xz
         zlib
         nettle
+        mbedtls
+        libxml2
     GIT_REPOSITORY https://github.com/libarchive/libarchive.git
     SOURCE_DIR ${SOURCE_LOCATION}
     GIT_CLONE_FLAGS "--filter=tree:0"
     UPDATE_COMMAND ""
-    CONFIGURE_COMMAND ${EXEC} autoreconf -fi && <SOURCE_DIR>/configure
-        --host=${TARGET_ARCH}
-        --prefix=${MINGW_INSTALL_PREFIX}
-        --disable-shared
-        --disable-bsdtar
-        --disable-bsdcat
-        --disable-bsdcpio
-        --with-lzo2
-        --without-xml2
-        --without-cng
-        --without-openssl
-    BUILD_COMMAND ${MAKE}
-    INSTALL_COMMAND ${MAKE} install
-    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND ${EXEC} cmake -H<SOURCE_DIR> -B<BINARY_DIR>
+        -G Ninja
+        -DCMAKE_INSTALL_PREFIX=${MINGW_INSTALL_PREFIX}
+        -DCMAKE_FIND_ROOT_PATH=${CMAKE_INSTALL_PREFIX}
+        -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}
```

Comment:
Put `DCMAKE_BUILD_TYPE` here before project's cmake option for consistency. Apply to all cmake build